tests/integration/client_v2_test.py

```diff
@@ -95,15 +77,24 @@ def test_whisper_v2_url_in_post(client_v2, data_dir, output_mode, mode, url, inp
 
     exp_basename = f"{Path(input_file).stem}.{mode}.{output_mode}.txt"
     exp_file = os.path.join(data_dir, "expected", exp_basename)
-    with open(exp_file, encoding="utf-8") as f:
+    # verify extracted text
+    do_fuzzy_assertion_with_extracted_text(exp_file, whisper_result, mode, output_mode)
+    usage_after = client_v2.get_usage_info()
+    # Verify usage after extraction
+    verify_usage(usage_before, usage_after, page_count, mode)
+
+
+def do_fuzzy_assertion_with_extracted_text(file_path, whisper_result, mode=None, output_mode=None):
+    with open(file_path, encoding="utf-8") as f:
         exp = f.read()
 
     assert isinstance(whisper_result, dict)
     assert whisper_result["status_code"] == 200
 
-    # For text based processing, perform a strict match
-    if mode == "native_text" and output_mode == "text":
-        assert whisper_result["extraction"]["result_text"] == exp
+    if mode and output_mode:
```

Comment:
@nagesh-zip is this condition mandatory? The fuzzy assertion will now be performed only when these values are `None` but it should ideally be done for OCR based